vimeo/VimeoNetworking
# Add initial support for SwiftLint

### Pull Request Checklist

### Ticket Summary
Add initial support for SwiftLint

### Implementation Summary
SwiftLint is being included in this project as a Pod, and will run during the script build phase for both the iOS and tvOS targets. 

We pass the `autocorrect` flag to automatically make changes to our source files to ensure consistency.

**Added SwiftLint to Podfile**
Added version 0.25.0 to the Podfile for both the iOS and tvOS targets.

**Enabled initial rules**
To start with only two rules will be autocorrected:
    - trailing_newline
    - trailing_whitespace

`trailing_newline` looks at the end of a file for extra newline characters. It ensures that only a single newline exists at the end of the file.

`trailing_whitespace` looks for extra whitespace at the end of a line. It then removes this whitespace.

These two rules were chosen for their low impact and lack of visible changes. More rules can be added in the future (for example, the `opening_brace` rule might be a fun one try to out 🤔)

### Reviewer Tips
If you want to take a look at what SwiftLint has to offer you can install it locally by running:
`brew install swiftlint`

If you want to see a list all the rules available run `swiftlint rules`.

Lastly, if you want to see the definition for a particular rule, and examples of what violations that will trigger an autocorrection, you can run `swiftlint rules opening_brace`.

### How to Test
- Build and run the example app.
- Build and run tests.
- Ensure that only whitepsace was removed.
